### ISSUE TITLE
Render scale circles to help to estimate distances

### ISSUE THREAD

**Issue by t-muko** [openlayers/openlayers#14168]
Understanding the map scale with zoomable map is sometimes difficult. Scale line or scale bar is the only hint available, but when using Openlayers with topographic maps, more intuitive scale illustration would be useful.

Having circles showing the distances would be more intuitive than just a line. 

![image](https://user-images.githubusercontent.com/18803269/193776510-46be887d-c1bf-46c9-a66a-a00c621afb58.png)

I am working an a solution to implement a ScaleCircles class. The implementation is based on the existing ScaleLine class, but functional requirements seem to differ enough to justify a separate class.

**Comment by t-muko** [comment]
@ahocevar The code is now completed and cleaned up. Waiting for the permission to submit my first contribution:
https://github.com/t-muko/openlayers/tree/scale-circles

**Comment by mike-000** [comment]
Depending on the projection and scale circles on the ground may not be circles in the view projection.  To get the correct shape you can draw to the vector context of a layer (either an existing layer or one created for this purpose) for example https://codesandbox.io/s/scalerings-forked-srro95

**Comment by t-muko** [comment]
> Depending on the projection and scale circles on the ground may not be circles in the view projection.

Good point Mike. The feature was developed to be used in smaller scales like with topographic maps, where the projection distortion is not a big source of error. Of course the world map is a different case. 

I will have a look on your example and also test the performance, which my current solution has some problems. Maybe we can tackle two issues with vector context.

**Comment by t-muko** [comment]
>  for example https://codesandbox.io/s/scalerings-forked-srro95
I am finally working on this feature. I noticed a weird behavior when rotating the map with circles and text on vectorContext. The text size and line width starts to decrease. After rotating 90 degrees, the ring becomes non-continuous! 
![image](https://user-images.githubusercontent.com/18803269/202391626-70818afc-522e-4611-8f8c-fadf4473fe5c.png)

**Comment by mike-000** [comment]
That looks like a bug in the vector context of tile layers (where the layer canvas is rotated).  It does not happen if you use a layer type where the canvas is not rotated (e.g. a vector layer) https://codesandbox.io/s/scalerings-forked-iub5kx?file=/index.js
